gobblin-hive-registration/src/main/java/gobblin/hive/HiveRegister.java

```diff
@@ -49,57 +40,45 @@
 import gobblin.hive.spec.activity.Activity;
 import gobblin.util.ExecutorsUtils;
 import gobblin.util.executors.ScalingThreadPoolExecutor;
+import lombok.Getter;
 import lombok.extern.slf4j.Slf4j;
 
 
 /**
- * A class for registering Hive tables or partitions.
- *
- * <p>
- *   An instance of this class is initialized with a {@link State} object. To register a table or partition,
- *   properties {@link HiveRegProps#HIVE_DB_ROOT_DIR}, {@link HiveRegProps#HIVE_OWNER} are required. They are
- *   optional for de-registering or checking the existence of a table or partition.
- * </p>
- *
- * <p>
- *   The {@link #register(HiveSpec)} method is asynchronous and returns immediately. Registration is performed in a
- *   thread pool whose size is controlled by {@link HiveRegProps#HIVE_REGISTER_THREADS}.
- * </p>
+ * A class for registering Hive tables and partitions.
  *
  * @author ziliu
  */
 @Slf4j
 @Alpha
-public class HiveRegister implements Closeable {
+public abstract class HiveRegister implements Closeable {
 
-  private static final String HIVE_DB_EXTENSION = ".db";
+  public static final String HIVE_REGISTER_TYPE = "hive.register.type";
+  public static final String DEFAULT_HIVE_REGISTER_TYPE = "gobblin.hive.metastore.HiveMetaStoreBasedRegister";
 
-  private final HiveRegProps props;
-  private final Optional<String> hiveDbRootDir;
-  private final GenericObjectPool<IMetaStoreClient> clientPool;
+  protected static final String HIVE_DB_EXTENSION = ".db";
 
-  private final ListeningExecutorService executor;
-  private final List<Future<Void>> futures = Lists.newArrayList();
-  private final HiveStripedLocks locks = new HiveStripedLocks();
+  @Getter
+  protected final HiveRegProps props;
 
-  public HiveRegister(State state) throws IOException {
+  protected final Optional<String> hiveDbRootDir;
+  protected final ListeningExecutorService executor;
+  protected final List<Future<Void>> futures = Lists.newArrayList();
+  protected final HiveStripedLocks locks = new HiveStripedLocks();
```

Comment:
are the locks needed when using the query based approach? https://cwiki.apache.org/confluence/display/Hive/Locking